src/sas/sascalc/data_util/registry.py

```diff
@@ -59,62 +62,56 @@ class ExtensionRegistry(object):
         registry.load('hello.cx',format='cx3') ->
             return cx3('hello.cx')
     """
-    def __init__(self, **kw):
-        self.loaders = {}
+    def __init__(self):
+        self.readers = defaultdict(list)
 
     def __setitem__(self, ext, loader):
-        if ext not in self.loaders:
-            self.loaders[ext] = []
-        self.loaders[ext].insert(0, loader)
+        self.readers[ext].insert(0, loader) # TODO: Why insert at zero, not just append?
 
     def __getitem__(self, ext):
-        return self.loaders[ext]
+        return self.readers[ext]
 
-    def __contains__(self, ext):
-        return ext in self.loaders
+    def __contains__(self, ext: str):
+        return ext in self.readers
 
     def formats(self):
         """
         Return a sorted list of the registered formats.
         """
-        names = [a for a in self.loaders.keys() if not a.startswith('.')]
+        names = [a for a in self.readers.keys() if not a.startswith('.')] # What is this doing?
```

Comment:
This is filtering out built-in readers from user-defined ones. It's possible for a user to associate a reader with another extension and user-defined associations should have no leading `.`. This is not currently exposed in the GUI.

Reply:
Thanks